### PR TITLE
[master] Local performance fixes

### PR DESCRIPTION
**Pull request by art-gor**
## Description

Minor performance improvements in internal messaging.

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/depends/safeserver/safehttpserver.cpp [hunk 1]
```diff
@@ -9,35 +9,44 @@
...
+      threads(threads),
+      running(false),
+      path_sslcert(sslcert),
+      path_sslkey(sslkey),
+      daemon(NULL),
+      bindlocalhost(false) {
+  if (threads < 1) {
+    this->threads = 1;
+  } else if (threads > 50) {
+    this->threads = 50;
```
Comment (valeryz) [inline src/depends/safeserver/safehttpserver.cpp:47]: Why are we capping threads at 50 here? How do we know this won't cause problems in prod if all threads get busy?

File: src/depends/safeserver/safehttpserver.cpp [hunk 2]
```diff
@@ -107,20 +114,20 @@ bool SafeHttpServer::StartListening() {
             SafeHttpServer::callback, this, MHD_OPTION_HTTPS_MEM_KEY,
             this->sslkey.c_str(), MHD_OPTION_HTTPS_MEM_CERT,
             this->sslcert.c_str(), MHD_OPTION_THREAD_POOL_SIZE, this->threads,
+            MHD_OPTION_LISTEN_BACKLOG_SIZE, LISTEN_BACKLOG_SIZE,
```
Comment (valeryz) [inline src/depends/safeserver/safehttpserver.cpp:117]: Could you add a comment about these new options. Why are they needed here and what they do?

File: src/libData/AccountData/AccountStore.cpp [hunk 3]
```diff
@@ -67,7 +67,8 @@ AccountStore::AccountStore() {
   }
   // EVM required to run on Lookup nodes too for view calls
   if (ENABLE_EVM) {
-    if (not ipcScillaInit && !LOOKUP_NODE_MODE) {
+    // TODO lookup nodes may also need it
+    if (not ipcScillaInit /*&& !LOOKUP_NODE_MODE*/) {
```
Comment (valeryz) [inline src/libData/AccountData/AccountStore.cpp:71]: This change seems wrong, as we should not change anything related to Scilla.

File: src/libDirectoryService/DirectoryService.cpp [hunk 4]
```diff
@@ -1189,6 +1189,11 @@ uint8_t DirectoryService::CalculateNewDifficultyCore(uint8_t currentDifficulty,
                                                      int64_t powSubmissions,
                                                      int64_t expectedNodes,
                                                      uint32_t powChangeoAdj) {
+  if (powChangeoAdj == 0) {
```
Comment (valeryz) [inline src/libDirectoryService/DirectoryService.cpp:1192]: Can you explain in the comments what this special mode for developers mean?

File: src/libNetwork/P2PComm.h [hunk 5]
```diff
@@ -19,7 +19,7 @@
 #define ZILLIQA_SRC_LIBNETWORK_P2PCOMM_H_
 
 #include <event2/util.h>
-#include <boost/lockfree/queue.hpp>
+// XXX #include <boost/lockfree/queue.hpp>
```
Comment (valeryz) [inline src/libNetwork/P2PComm.h:22]: Please remove this.

File: src/libNetwork/P2PComm.h [hunk 6]
```diff
@@ -114,8 +115,11 @@ class P2PComm {
 
   ThreadPool m_SendPool{MAXSENDMESSAGE, "SendPool"};
 
-  boost::lockfree::queue<SendJob*> m_sendQueue;
-  void ProcessSendJob(SendJob* job);
+  // TODO shared instead of unique due to lambda move capture limitations
```
Comment (valeryz) [inline src/libNetwork/P2PComm.h:118]: TODO comment seems not needed.

**Review by valeryz — COMMENTED** [review]
A nice change!

But please address all the comments, and get @CSideSteve feedback.

File: src/libData/AccountData/AccountStore.cpp (hunk 3 above)
Comment (CSideSteve) [reply src/libData/AccountData/AccountStore.cpp:71]: Changes in this area were introduced for the work Kaustubh was doing on the API interface.

We have some states that the logic here doesn't quit match, these are :

Running as a Zilliqa process and not the Lookup Node or LOOKUP_NODE_MODE setting

Running as the isolatedServer

Running the testroutines as part of the build causes an invocation of this method and has requirements too.

Comment (valeryz) [reply src/libData/AccountData/AccountStore.cpp]: Kaustub's change were to enable EVM on seeds/lookups.  But Scilla should not be affected.

**Review by CSideSteve — APPROVED** [review]
Looks Good

**Pull request merged**